schneiderWallbox/integrationpluginschneiderwallbox.cpp

```diff
@@ -0,0 +1,546 @@
+            ParamTypeId macAddressParamType = thingClass.paramTypes().findByName("mac").id();
+
+            ThingDescriptor descriptor(info->thingClassId(), thingClass.displayName(), networkDeviceInfo.address().toString());
+            descriptor.setParams({Param(macAddressParamType, networkDeviceInfo.macAddress())});
+
+            // Check if we already have set up this device
+            Thing *existingThing = myThings().findByParams(descriptor.params());
+            if (existingThing) {
+                qCDebug(dcSchneiderElectric()) << "Found already existing" << thingClass.name() << "wallbox:" << existingThing->name() << networkDeviceInfo;
+                descriptor.setThingId(existingThing->id());
+            } else {
+                qCDebug(dcSchneiderElectric()) << "Found new" << thingClass.name() << "wallbox";
+            }
+
+            info->addThingDescriptor(descriptor);
+        }
+        info->finish(Thing::ThingErrorNoError);
+    });
+}
+
+
+void IntegrationPluginSchneiderWallbox::setupThing(ThingSetupInfo *info)
+{
+    Thing *thing = info->thing();
+
+    /* Probably not needed.
+    if (thing->thingClassId() != schneiderEVlinkThingClassId) {
+        qCWarning(dcSchneiderElectric()) << "Could not setup thing: unhandled device class" << thing->thingClass();
+        info->finish(Thing::ThingErrorThingClassNotFound);
+        return;
+    }
+    */
+
+    if (m_schneiderDevices.contains(thing)) {
+        qCDebug(dcSchneiderElectric()) << "Reconfiguring existing thing" << thing->name();
+        m_schneiderDevices.take(thing)->deleteLater();
+    } else {
+        qCDebug(dcSchneiderElectric()) << "Setting up a new device:" << thing->params();
+    }
+
+    MacAddress mac = MacAddress(thing->paramValue("mac").toString());
+    if (!mac.isValid()) {
+        info->finish(Thing::ThingErrorInvalidParameter, QT_TR_NOOP("The given MAC address is not valid."));
+        return;
+    }
+    NetworkDeviceMonitor *monitor = hardwareManager()->networkDeviceDiscovery()->registerMonitor(mac);
+
+    uint port = thing->paramValue("port").toUInt();
+    quint16 slaveId = thing->paramValue("slaveId").toUInt();
+    SchneiderWallboxModbusTcpConnection *schneiderWallboxTcpConnection = new SchneiderWallboxModbusTcpConnection(monitor->networkDeviceInfo().address(), port, slaveId, this);
+    SchneiderWallbox *schneiderWallbox = new SchneiderWallbox(schneiderWallboxTcpConnection, this);
+    connect(info, &ThingSetupInfo::aborted, schneiderWallbox, &SchneiderWallbox::deleteLater);
+    connect(info, &ThingSetupInfo::aborted, monitor, [monitor, this](){ hardwareManager()->networkDeviceDiscovery()->unregisterMonitor(monitor);});
+
+    // ToDO: Check if this is really a Schneider wallbox.
+
+    quint16 minCurrentLimit = thing->paramValue("minChargeCurrent").toUInt();
+    quint16 maxCurrentLimit = thing->paramValue("maxChargeCurrent").toUInt();
+    thing->setStateMinMaxValues("maxChargingCurrent", minCurrentLimit, maxCurrentLimit);
```

Comment:
Note for myself: Depending on the outcome of the min/max discussion in the .json, this needs to be re-evaluated.